src/main/java/io/github/thebusybiscuit/slimefun4/implementation/setup/ResearchSetup.java

```diff
@@ -283,6 +284,7 @@ public static void setupResearches() {
         register("wise_talisman", 271, "Talisman of the Wise", 20, SlimefunItems.TALISMAN_WISE);
         register("book_binder", 272, "Enchantment Book Binding", 26, SlimefunItems.BOOK_BINDER);
         register("auto_crafting", 273, "Automatic Crafting", 30, SlimefunItems.VANILLA_AUTO_CRAFTER, SlimefunItems.ENHANCED_AUTO_CRAFTER);
+        register("improved_generators", 274, "Improved Generators", 38, SlimefunItems.COAL_GENERATOR_2, SlimefunItems.LAVA_GENERATOR_2);
```

Comment:
I think 38 is hella overpriced as a default value.

```suggestion
        register("improved_generators", 274, "Improved Generators", 24, SlimefunItems.COAL_GENERATOR_2, SlimefunItems.LAVA_GENERATOR_2);
```

Reply:
Lava gen research is 38, I made it same cost. Should I lower the tier 1 lava gen then?

Reply:
I'll take it as a yes